src/app/ubs/ubs/components/ubs-pick-up-service-pop-up/ubs-pick-up-service-pop-up.component.ts

```
@@ -2,15 +1,120 @@
+      });
+  }
+
+  loadLocations(): void {
+    const courierId = this.courierUBS.courierId;
+    this.orderService
+      .getLocations(courierId, true)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((response: AllActiveLocationsDtosResponse) => {
+        this.cities = response.allActiveLocationsDtos.reduce(
+          (acc, region) => [
+            ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)